easybuild/tools/filetools.py

```diff
@@ -1628,6 +1629,65 @@ def copy(paths, target_path, force_in_dry_run=False):
             raise EasyBuildError("Specified path to copy is not an existing file or directory: %s", path)
 
 
+def get_source_from_git(filename, targetdir, git_config):
+    """
+    Downloads a git repository, at a specific tag or commit, recursively or not, and make an archive with it
+
+    :param filename: name of the archive to save the code to (must be .tar.gz)
+    :param targetdir: target directory where to save the archive to
+    :param git_config: dictionary containing url, repo_name, recursive, and one of tag or commit
+    """
+    # if a non-empty dictionary was provided, download from git and archive
+    if not isinstance(git_config, dict):
+        raise EasyBuildError("Found a non null git_config in 'sources', but value is not a dictionary")
+
+    # Making a copy to avoid modifying the object with pops
+    git_config = git_config.copy()
+    tag = git_config.pop('tag', None)
+    url = git_config.pop('url', None)
+    repo_name = git_config.pop('repo_name', None)
+    commit = git_config.pop('commit', None)
+    recursive = git_config.pop('recursive', False)
+    if git_config:
+        raise EasyBuildError("Found one or more unexpected keys in 'git_config' specification: %s", git_config)
+    if not repo_name:
+        raise EasyBuildError("repo_name not specified in git_config parameter")
+    if not tag and not commit:
+        raise EasyBuildError("Neither tag nor commit found in git_config parameter")
+    if tag and commit:
+        raise EasyBuildError("Tag and commit are mutually exclusive in git_config parameter")
+    if not url:
+        raise EasyBuildError("url not specified in git_config parameter")
+    if '.tar.gz' not in filename:
+        raise EasyBuildError("git_config only supports filename ending in .tar.gz")
+
+    mkdir(targetdir, parents=True)
+    targetpath = os.path.join(targetdir, filename)
+
+    cwd = change_dir(targetdir)
+    recursive = " --recursive " if recursive else ""
+    if tag:
+        cmd = "git clone --branch %s %s %s/%s.git " % (tag, recursive, url, repo_name)
+    else:
+        cmd = "git clone %s %s/%s.git" % (recursive, url, repo_name)
+    (cmdstdouterr, ec) = run.run_cmd(cmd, log_all=True, log_ok=False, simple=False, regexp=False)
```

Comment:
@mboisson You're not doing anything with `cmdstdouterr` or `ec` here, so I would let `run_cmd` check the exit code instead?

```python
run.run_cmd(cmd, log_all=True, log_ok=False, regexp=False)
```